elyra/pipeline/component.py

```diff
@@ -205,63 +186,67 @@ def __init__(self, id: str, name: str, description: Optional[str], source_type:
         self._op = op
         self._properties = properties
 
-        # Error checking for extensions & filehandler_parameter_ref depends on the component source type
-        if self._source_type == "elyra":
-            if not extensions:
-                raise ValueError("Invalid generic component: Missing field 'extensions'.")
+        if self._source_type == "elyra" and not filehandler_parameter_ref:
             filehandler_parameter_ref = "filename"
-        else:
-            if extensions and not filehandler_parameter_ref:
-                raise ValueError("Invalid component: A component for which file extensions are \
-                                 supplied must include the parameter_ref of the first parameter \
-                                 that is associated with that extensinon.")
+
+        if extensions and not filehandler_parameter_ref:
+            Component._log_warning(f"Component '{self._id}' specifies extensions '{extensions}' but \
+                                   no 'filehandler_parameter_ref' value and cannot participate in \
+                                   drag and drop functionality as a result.")
 
         self._extensions = extensions
         self._filehandler_parameter_ref = filehandler_parameter_ref
 
     @property
-    def id(self):
+    def id(self) -> str:
         return self._id
 
     @property
-    def name(self):
+    def name(self) -> str:
         return self._name
 
     @property
-    def description(self):
+    def description(self) -> str:
```

Comment:
Looks like description is optional, so this should be `-> Optional[str]`

Reply:
Not sure how I missed that!